addons/hr_holidays/models/hr_leave.py

```diff
@@ -73,11 +73,14 @@ def default_get(self, fields_list):
         defaults = super(HolidaysRequest, self).default_get(fields_list)
         defaults = self._default_get_request_parameters(defaults)
 
-        LeaveType = self.env['hr.leave.type'].with_context(employee_id=defaults.get('employee_id'), default_date_from=defaults.get('date_from', fields.Datetime.now()))
-        lt = LeaveType.search([('valid', '=', True)], limit=1)
+        if 'holidays_status_id' in fields_list and not defaults.get('holidays_status_id'):
+            lt = self.env['hr.leave.type'].search([('valid', '=', True)], limit=1)
 
-        defaults['holiday_status_id'] = lt.id if lt else defaults.get('holiday_status_id')
-        defaults['state'] = 'confirm' if lt and lt.leave_validation_type != 'no_validation' else 'draft'
+            defaults['holiday_status_id'] = lt.id if lt else defaults.get('holiday_status_id')
```

Comment:
you just checked `not defaults.get('holidays_status_id')` above so the `else` close will always be `None`